shell/key-bindings.zsh

```diff
@@ -106,6 +106,9 @@ bindkey '\ec' fzf-cd-widget
 fzf-history-widget() {
   local selected num
   setopt localoptions noglobsubst noposixbuiltins pipefail no_aliases 2> /dev/null
+  if [[ -o sharehistory && -z "$BUFFER" ]]; then
+    fc -R "$HISTFILE"
```

Comment:
Isn't `fc -R` sufficient?

Excerpt from http://zsh.sourceforge.net/Doc/Release/Shell-Builtin-Commands.html

> ‘fc -R’ reads the history from the given file, ‘fc -W’ writes the history out to the given file, and ‘fc -A’ appends the history out to the given file. If no filename is specified, the $HISTFILE is assumed.

Reply:
Looks like `[[ -o sharehistory ]] && fc -R` is all we need, but please correct me if I'm missing something.

Reply:
It prevents reloading history if the edit buffer is non-empty. E.g you typed something before pressing C-r, it's not strictly necessary though.

Reply:
And yes, looks like the "$HISTFILE" parameter can be omitted.